File: Chapter 03/JS/section_3_3_1_srp/after/playerRepository.js
/**
 * Responsibility 3: Data Persistence.
 * * ARCHITECTURE NOTE: This class handles all infrastructure concerns. If we 
 * move from a local MongoDB to a cloud-based PostgreSQL, or even just a 
 * JSON file for local development, this is the ONLY file that needs to change. 
 * * The rest of the game's logic remains completely unaware of how the data 
 * is actually stored.
 */
class PlayerRepository {
    /**
     * Persists the player's statistics to the underlying data store.
     * @param {Player} player - The player whose stats need saving.
     */
    saveStats(player) {
        console.log(`  [Database] Saving ${player.name}'s game stats to the database.`);
    }
}

module.exports = PlayerRepository;

File: Chapter 03/C#/Section_3_3_1_SRP/After/PlayerRepository.cs
using System;

namespace Chapter03.SRP.After;

/// <summary>
/// Responsibility 3: Data Persistence.
/// 
/// ARCHITECTURE NOTE: This class handles all database infrastructure. If we decide 
/// to migrate from a SQL database to a NoSQL database, or if our ORM framework 
/// changes, this is the *only* class that needs to be updated. The Player and 
/// Tactics classes remain completely untouched.
/// </summary>
public class PlayerRepository
{
    /// <summary>
    /// Persists the player's statistics to the underlying data store.
    /// </summary>
    /// <param name="player">The player whose stats need saving.</param>
    public void SaveStats(Player player) 
    {
        Console.WriteLine($"  [Database] Saving {player.Name}'s game stats to the database.");
    }
}